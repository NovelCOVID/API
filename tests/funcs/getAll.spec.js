const axios = require('axios')
const redis = require('ioredis')

const fixture = require('../worldometer-countries-fixture')
const getAll = require('../../funcs/getAll')

jest.mock('ioredis')
jest.mock('axios')
redis.set = jest.fn()

describe('getAll API', () => {
  beforeEach(() => {
    axios.get.mockClear()
    redis.set.mockClear()
  })
  it('scrapes for data', async () => {
    axios.get.mockResolvedValue({ data: fixture, status: 200 })

    const result = await getAll([], redis)
    const { updated, ...rest } = result

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(redis.set).toHaveBeenCalledTimes(1)
    expect(rest).toMatchSnapshot()
    expect(result).toMatchObject({
      cases: 399093,
      deaths: 17365,
      recovered: 103748,
    })
  })

  it('handles empty response from worldometer', async () => {
    axios.get.mockResolvedValue({ data: '', status: 200 })

    const result = await getAll([], redis)
    const { updated, ...rest } = result

    expect(axios.get).toHaveBeenCalledTimes(1)
    expect(redis.set).toHaveBeenCalledTimes(1)
    expect(rest).toMatchSnapshot()
    expect(Object.keys(result)).toEqual(['updated'])
  })
})
