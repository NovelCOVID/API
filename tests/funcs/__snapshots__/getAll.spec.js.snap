// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getAll API handles empty response from worldometer 1`] = `Object {}`;

exports[`getAll API scrapes for data 1`] = `
Object {
  "cases": 399093,
  "deaths": 17365,
  "recovered": 103748,
}
`;
